eslint/babel-eslint-parser/src/client.cjs

```diff
@@ -72,11 +72,29 @@ exports.WorkerClient = class WorkerClient extends Client {
         [subChannel.port1],
       );
 
-      Atomics.wait(this.#signal, 0, 0);
-      const { message } = WorkerClient.#worker_threads.receiveMessageOnPort(
-        subChannel.port2,
-      );
+      let response;
+      let i = 0;
+      // Sometimes receiveMessageOnPort returns "undefined" instead of the
+      // actual response object. Try multiple times, with a timeout of 5ms
+      // on Atomic.wait starting from the second one.
+      do {
+        const wakeReason = Atomics.wait(
+          this.#signal,
+          0,
+          i === 0 ? Infinity : 5,
+        );
```

Comment:
I think only the first time will work fine because `#signal` is not reset.

Reply:
I though the same, but your original commits did not reset `#signal` either and they seemed to work.

Reply:
Because I used another index to do sleep. :)
https://github.com/babel/babel/pull/14541#discussion_r870656401

Reply:
Whoops thanks, I didn't notice it